x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -275,24 +355,24 @@ static void getMemoryAndProcessors(
             }
 
             // min(3, max(number of allocations over all deployed models)
-            minNodes = Math.min(3, Math.max(minNodes, numberOfAllocations));
+            minNodes = Math.min(3, Math.max(minNodes, numberOfRequestedAllocations));
```

Comment:
I'm still not sure why we are selecting minNodes like this. The 3 seems arbitrary, and I don't understand why the number of allocations has anything to do with the min number of nodes. I guess this could have to do with availability zones?

Reply:
@edsavage do you know anything about minNodes?